96.md

```diff
@@ -0,0 +1,221 @@
+NIP-96
+======
+
+HTTP File Storage Integration
+-----------------------------
+
+`draft` `optional` `author:arthurfranca` `author:Semisol` `author:staab`
+
+## Introduction
+
+This NIP defines a REST API for HTTP file storage servers intended to be used in conjunction with the nostr network.
+The API will enable nostr users to upload files and later reference them by url on nostr notes.
+
+The spec DOES NOT use regular nostr events through websockets for
+storing, requesting nor retrieving data because, for simplicity, the server
+will not have to learn anything about nostr relays.
+
+## Server Adaptation
+
+File storage servers wishing to be accessible by nostr users should opt-in by making available an https route at `/.well-known/nostr/nip96.json` with `api_url`:
+
+```js
+{
+  // Required
+  // File upload and deletion are served from this url
+  // Also downloads if "download_url" field is absent or empty string
+  "api_url": "https://your-file-server.example/custom-api-path",
+  // Optional
+  // If absent, downloads are served from the api_url
+  "download_url": "https://a-cdn.example/a-path",
```

Comment:
> separation of endpoints based on a file type. If the target just a file and not media, then this may be OK.

There are libs able to detect file type just from the begginning of a file content like [https://github.com/sindresorhus/file-type](https://github.com/sindresorhus/file-type). Can't you use this to block upload of unsupported types or types that weren't able to be recognized by such a lib?

Reply:
Yeah sorry downloads. But it is a fixed download path that the client will just append the file hash it wants to download.

There will be also temporary download urls for cases of waiting for server media processing as @quentintaranpino just suggested. But these will be inserted in the upload response instead of in this config file.

Reply:
I will expand this topic at your conversation below.